django/test/runner.py

```diff
@@ -283,17 +284,23 @@ def run(self, test):
         return result
 
 
-def default_test_processes():
-    """Default number of test processes when using the --parallel option."""
+def parallel_type(string):
+    """Parse value passed to the --parallel option."""
     # The current implementation of the parallel test runner requires
     # multiprocessing to start subprocesses with fork().
     if multiprocessing.get_start_method() != 'fork':
         return 1
-    try:
-        return int(os.environ['DJANGO_TEST_PROCESSES'])
-    except KeyError:
+
+    if string == "auto":
         return multiprocessing.cpu_count()
 
+    try:
+        return int(string)
+    except ValueError:
+        raise argparse.ArgumentTypeError(
+            "{} is not an integer or the string 'auto'".format(repr(string))
```

Comment:
We could use `{!r}` here to remove the need for manually calling `repr(string)`?

Reply:
I can even use an f-string since we support Python 3.6+ only now! It looks like it will be the first in the code base though??

Reply:
Historic moment!